### PR TITLE
Add a quick exit widget and change the default configuration.

### PR DESCRIPTION
**Pull request by sux2mfgj**
Some times I failed to change my configuration file of Qtile. At the time, the default configuration was loaded. but, From the qtile with the default configuration, I couldn't shut down the qtile, because of a key map of the command doesn't find easily. 

In order to resolve the problem, I implemented a quick exit widget using the mouse (GUI) and added the widget to the default configuration loaded on error detected.

### CODE REVIEW

**Comment by ramnes** [comment]
Thanks for the PR! Why the countdown?

**Comment by sux2mfgj** [comment]
Thank you for your comment.

I want to prevent qtile shut down accidentally.

**Comment by sux2mfgj** [comment]
I implemented the countdown style, but there are any styles. How do you think about it?

File: libqtile/widget/quick_exit.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from enum import Enum
+
+from . import base
+from .. import bar
+
+
+class State(Enum):
```
Comment (flacjacket) [inline libqtile/widget/quick_exit.py]: You can just use `auto()` for the values here.

File: libqtile/widget/quick_exit.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from enum import Enum
+
+from . import base
+from .. import bar
```
Comment (flacjacket) [inline libqtile/widget/quick_exit.py]: I know this is done throughout, but I feel it is better to do a fully qualified import from `libqtile`

File: libqtile/widget/quick_exit.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        self.draw()
+
+    def button_press(self, x, y, button):
+        if self.state == State.Neutral:
+            self.state = State.Counting
+            self.update()
+            return
+
+        if self.state == State.Counting:
+            self.qtile.stop()
```
Comment (flacjacket) [inline libqtile/widget/quick_exit.py]: Shouldn't a button press while it is counting reset it back to neutral?

Comment (ramnes) [reply libqtile/widget/quick_exit.py]: My thoughts also

File: libqtile/widget/quick_exit.py (hunk 1 above)
Comment (sux2mfgj) [reply libqtile/widget/quick_exit.py]: thanks. I change to use auto.

File: libqtile/widget/quick_exit.py (hunk 2 above)
Comment (sux2mfgj) [reply libqtile/widget/quick_exit.py]: I fixed it.

File: libqtile/widget/quick_exit.py (hunk 3 above)
Comment (sux2mfgj) [reply libqtile/widget/quick_exit.py]: I agree with your comments. I changed this behavior.
https://github.com/qtile/qtile/pull/1389/commits/dba37a6bd5a06710f2899d995de5df65f205654e

File: libqtile/widget/quick_exit.py (hunk 1 above)
Comment (sux2mfgj) [reply libqtile/widget/quick_exit.py]: https://github.com/qtile/qtile/pull/1389/commits/cbee326059f76d30451bc02245a4aa3229fc4ceb

**Comment by sux2mfgj** [comment]
There is no enum.auto at python 3.5, so the test failed. Do we have to use any library that can fill the gap?  e.g. [aenum](https://pypi.org/project/aenum/).

**Comment by sux2mfgj** [comment]
I removed enum and use boolean instead.

**Comment by flacjacket** [comment]
Great, thanks for this!

**Pull request merged**